src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdLocationUpdateService.java

```diff
@@ -0,0 +1,111 @@
+package uk.gov.hmcts.reform.sscs.service.ccdupdate;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Address;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Hearing;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Venue;
+import uk.gov.hmcts.reform.sscs.exception.UpdateCaseException;
+import uk.gov.hmcts.reform.sscs.model.VenueDetails;
+import uk.gov.hmcts.reform.sscs.model.messaging.HmcMessage;
+import uk.gov.hmcts.reform.sscs.service.venue.VenueRpcDetails;
+import uk.gov.hmcts.reform.sscs.service.venue.VenueRpcDetailsService;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.isNull;
+import static java.util.stream.Collectors.toList;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CcdLocationUpdateService {
+
+    private final VenueRpcDetailsService venueRpcDetailsService;
+
+    public void updateVenue(HmcMessage hmcMessage, SscsCaseData sscsCaseData) throws UpdateCaseException {
+
+        Optional<HearingDetails> hearingDetails = getHearingDetailsFromHearingList(hmcMessage, sscsCaseData);
+        if (hearingDetails.isEmpty()) {
+
+            UpdateCaseException exc = new UpdateCaseException(
+                String.format("Failed to update venue in CCD(Case Id: %s) - "
+                                  + "can not find hearing with Id: %s", sscsCaseData.getCcdCaseId(), hmcMessage.getHearingID()));
```

Comment:
As a comment only for future dev, better to avoid splitting strings unless it goes beyond the line (I don't think it would here):
                String.format("Failed to update venue in CCD(Case Id: %s) - can not find hearing with Id: %s",
                   sscsCaseData.getCcdCaseId(), hmcMessage.getHearingID()));

Reply:
Because of the Checkstyle, per-line has character length limitation. This branch created by Rokas at early of Apirl, and the length limitation was 130, that's why we splitting string like this. Yes, we will not splitting string like this, since the length limitation increased to 240